lib/surface/compiler/eex_engine.ex

```diff
@@ -219,16 +219,41 @@ defmodule Surface.Compiler.EExEngine do
 
     dynamic_props_expr = handle_dynamic_props(dynamic_props)
 
-    {case_expression, slot_meta, slot_props} =
-      collect_slot_meta(component, templates, buffer, state)
+    if module.__use_context__? do
+      Module.put_attribute(meta.caller.module, :use_context?, true)
```

Comment:
we should probably move this attribute setting into the compiler

Reply:
Besides this one, there are other things related to contexts that we need to decide the best place to put them. With the current implementation, those things are spread along `Context`, `EExEngine`, `ContextPut` and `ContextGet`. Since we now have everything working, we can start thinking about refactoring the whole thing. Maybe for `v0.2.0`.